demonstrations/tutorial_kernel_module.py

```diff
@@ -0,0 +1,596 @@
+    tutorial_data_reuploading_classifier Classification with data reuploading
+
+*Authors: Peter-Jan Derks, Paul Fährmann, Elies Gil-Fuster, Tom
+Hubregtsen, Johannes Jakob Meyer and David Wierichs. Posted: 24 June 2021*
+
+Kernel methods are one of the cornerstones of classical machine
+learning. To understand what a kernel method does, let’s first revisit
+one of the simplest methods to assign binary labels to datapoints:
+linear classification.
+
+Imagine we want to discern two different classes of points that lie in
+different corners of the plane. A linear classifier corresponds to
+drawing a line and assigning different labels to the regions on opposing
+sides of the line:
+
+.. figure:: ../demonstrations/kernel_module/linear_classification.png
+   :align: center
+   :width: 30%
+
+We can mathematically formalize this by assigning the label :math:`y`
+via
+
+.. math::
+   y(\boldsymbol{x}) = \operatorname{sgn}(\langle \boldsymbol{w}, \boldsymbol{x}\rangle + b).
+
+The vector :math:`\boldsymbol{w}` points perpendicular to the line and
+thus determine its slope. The independent term :math:`b` specificies the
+position on the plane. In this form, linear classification can also be
+extended to higher dimensional vectors :math:`\boldsymbol{x}`, where a
+line does not divide the entire space into two regions anymore. Instead
+one needs a *hyperplane*. It is immediately clear that this method is
+not very powerful, as datasets that are not separable by a hyperplane
+can’t be treated.
+
+We can actually sneak around this limitation by performing a neat trick:
+if we define some map :math:`\phi(\boldsymbol{x})` that *embeds* our
+datapoints into a larger *feature space* and then perform linear
+classification there, we could actually realise non-linear
+classification in our original space!
+
+.. figure:: ../demonstrations/kernel_module/embedding_nonlinear_classification.png
+   :align: center
+   :width: 65%
+
+If we go back to the expression for our prediction and include the
+embedding, we get
+
+.. math::
+   y(\boldsymbol{x}) = \operatorname{sgn}(\langle \boldsymbol{w}, \phi(\boldsymbol{x})\rangle + b).
+
+We will forgo one tiny step, but it can be shown that for the purposes
+of optimal classification, we can choose the vector defining the
+decision boundary as a linear combination of the embedded datapoints
+:math:`\boldsymbol{w} = \sum_i \alpha_i \phi(\boldsymbol{x}_i)`. Putting
+this into the formula yields
+
+.. math::
+   y(\boldsymbol{x}) = \operatorname{sgn}\left(\sum_i \alpha_i \langle \phi(\boldsymbol{x}_i), \phi(\boldsymbol{x})\rangle + b\right).
+
+This rewriting might not seem useful at first, but notice the above
+formula only contains inner products between vectors in the embedding
+space:
+
+.. math::
+   k(\boldsymbol{x}_i, \boldsymbol{x}_j) = \langle \phi(\boldsymbol{x}_i), \phi(\boldsymbol{x}_j)\rangle.
+
+We call this function the *kernel*. The clue now is that we can often
```

Comment:
```suggestion
We call this function the *kernel*. The important point is that we can often
```

Lol, I _think_ this is an exclusively German formulation.

Reply:
Good catch :D